### DISCUSSION TITLE
starting host-side gRPC server extremely slow

### DISCUSSION THREAD

**Discussion by christiansalway1** [terraform-linters/tflint#1578] [Q&A] [answered]
It's taking 9 minutes to run tflint against our small sized terraform.  When I run `TFLINT_LOG=debug tflint` it takes most of that time starting host-side gRPC server
```
13:02:50 [DEBUG] go-plugin@v1.4.5/client.go:736: using plugin: version=11
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:08:14 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
13:08:14 server.go:148: [INFO] unknown value found in ../../modules/agw_to_lambda/api_gateway.tf:41. TFLint can only evaluate provided variables and skips dynamic values.
13:08:14 [ERROR] interceptor/logging.go:18: failed to gRPC request: direction=plugin2host method=/proto.Runner/EvaluateExpr err="rpc error: code = FailedPrecondition desc = unknown value found in ../../modules/agw_to_lambda/api_gateway.tf:41"
```

```
% tflint -v
TFLint version 0.42.2
+ ruleset.terraform (0.2.1-bundled)
```

```
#.tflint.hcl
plugin "terraform" {
    enabled = true
    version = "0.2.1"
    source  = "github.com/terraform-linters/tflint-ruleset-terraform"
}
plugin "aws" {
    enabled = true
    version = "0.18.0"
    source  = "github.com/terraform-linters/tflint-ruleset-aws"
}
config {
  module = true
}

**Comment by wata727** [comment]
It may be because modules have been expanded since v0.42. Until now, only 1 module was inspected for the following config, but from v0.42, 10 modules will be inspected.

```hcl
module "aws_instance" {
  count = 10
}
```

In the module inspection, TFLint starts a gRPC server for each module, so such logs may appear. Please tell me about the following:

- Do you enable module inspection with `--module` or something like that?
- Do you have a module definition with a large number specified in `count` or `for_each`?

**Comment by wata727** [comment, marked as answer]
We noticed that Terraform ruleset v0.2.1 has some performance issues. You may be affected by this. See also https://github.com/terraform-linters/tflint-ruleset-terraform/pull/49.

This will be fixed in the next version. Current workaround is to downgrade to v0.1.1.

**Reply by wata727** [reply]
Terraform ruleset v0.2.2 has been released. This version fixes the performance issue.
TFLint v0.44.0 bundles this version. Please try here.

**Comment by RohanUB121** [comment]
> It's taking 9 minutes to run tflint against our small sized terraform. When I run `TFLINT_LOG=debug tflint` it takes most of that time starting host-side gRPC server
> 
> ```
> 13:02:50 [DEBUG] go-plugin@v1.4.5/client.go:736: using plugin: version=11
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:02:50 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:08:14 [DEBUG] host2plugin/client.go:124: starting host-side gRPC server
> 13:08:14 server.go:148: [INFO] unknown value found in ../../modules/agw_to_lambda/api_gateway.tf:41. TFLint can only evaluate provided variables and skips dynamic values.
> 13:08:14 [ERROR] interceptor/logging.go:18: failed to gRPC request: direction=plugin2host method=/proto.Runner/EvaluateExpr err="rpc error: code = FailedPrecondition desc = unknown value found in ../../modules/agw_to_lambda/api_gateway.tf:41"
> ```
> 
> ```
> % tflint -v
> TFLint version 0.42.2
> + ruleset.terraform (0.2.1-bundled)
> ```
> 
> ```
> #.tflint.hcl
> plugin "terraform" {
>     enabled = true
>     version = "0.2.1"
>     source  = "github.com/terraform-linters/tflint-ruleset-terraform"
> }
> plugin "aws" {
>     enabled = true
>     version = "0.18.0"
>     source  = "github.com/terraform-linters/tflint-ruleset-aws"
> }
> config {
>   module = true
> }
> ```

Hey @christiansalway1 I am facing similar issue with Tflint as shown in the last two lines  in your screenshot. Unknown value found in ../..modules....
Error interceptor/logging.go:18 failed to gRPC request. Were you able to get those resolved? Any help is appreciated thanks.
